src/main/java/net/neoforged/neoforge/network/registration/NetworkChannel.java

```diff
@@ -16,6 +18,9 @@
  * @param chosenVersion The chosen version, if any.
  */
 @ApiStatus.Internal
-public record NetworkChannel(
-        ResourceLocation id,
-        Optional<String> chosenVersion) {}
+public record NetworkChannel(ResourceLocation id, String chosenVersion) {
```

Comment:
where did the optional go? are versions mandatory now?

Reply:
Yes, though we have noted it will not make people use meaningful versions, there isn't any distinction between a useless version and an `Optional.empty()`, and the code flow is easier without having to deal with the `Optional`.

Reply:
Potentially we can add a `registrar` method that defaults to a `""` version.

Reply:
Electing to leave as-is; network versions are probably going to be less relevant in the long term as we roll out the handshake protocol.